### PR TITLE
Add /api/labels endpoint

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Resolves https://github.com/web-platform-tests/wpt.fyi/issues/329

Adds a endpoint /api/labels that lists all of the labels used on any runs.

Loads `TestRun` entities with all distinct `Labels`, aggregates them into a set, spits out a sorted array, caches the whole request for 5 min.

### CODE REVIEW

File: api/labels.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	mapset "github.com/deckarep/golang-set"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+)
+
+// LabelsHandler is an http.Handler for the /api/labels endpoint.
+type LabelsHandler struct {
+	ctx context.Context
+}
+
+// apiLabelsHandler is responsible for emitting just the revision SHAs for test runs.
```
Comment (foolip) [inline api/labels.go]: Is this comment copypasta? It emits labels, I hope :)

File: api/labels.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	store := shared.NewAppEngineDatastore(h.ctx)
+	var runs shared.TestRuns
+	_, err := store.GetAll(store.NewQuery("TestRun").Project("Labels").Distinct(), &runs)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	all := mapset.NewSet()
+	for _, run := range runs {
```
Comment (foolip) [inline api/labels.go]: Is this really able to iterate all runs, won't this run into some limit that requires pagination just like /api/runs?

File: api/labels_medium_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	"google.golang.org/appengine/datastore"
+)
+
+func TestLabelsHandler(t *testing.T) {
+	ctx, done, err := sharedtest.NewAEContext(true)
+	assert.Nil(t, err)
+	defer done()
+
+	key := datastore.NewIncompleteKey(ctx, "TestRun", nil)
+	_, err = datastore.Put(ctx, key, &shared.TestRun{Labels: []string{"a"}})
```
Comment (foolip) [inline api/labels_medium_test.go]: Can you add the labels out of order to also test the sorting, and perhaps also a duplicate?

File: api/labels.go (hunk 1 above)
Comment (lukebjerring) [reply api/labels.go]: Sure was.

File: api/labels.go (hunk 2 above)
Comment (lukebjerring) [reply api/labels.go]: My theory (seems to work in staging, which has more labels than prod) is that the `.Project("Labels").Distinct()` part means that we get less than 500 distinct results, since only distinct sets of labels are included in the result.

File: api/labels_medium_test.go (hunk 3 above)
Comment (lukebjerring) [reply api/labels_medium_test.go]: Done.
